src/Features/Core/Portable/Completion/CompletionService.cs

```diff
@@ -39,6 +51,14 @@ internal CompletionService()
         public static CompletionService? GetService(Document? document)
             => document?.GetLanguageService<CompletionService>();
 
+        /// <summary>
+        /// Returns the providers always available to the service.
+        /// This does not included providers imported via MEF composition.
+        /// </summary>
+        [Obsolete("Built-in providers will be ignored in a future release, please make them MEF exports instead.")]
+        protected virtual ImmutableArray<CompletionProvider> GetBuiltInProviders()
+            => ImmutableArray<CompletionProvider>.Empty;
```

Comment:
does anyone override this?  can we remove now (or in a quick followup pr?)

Reply:
Both F# and TS are still using this, unfortunately
https://github.com/dotnet/fsharp/blob/main/vsintegration/src/FSharp.Editor/Completion/CompletionService.fs
https://devdiv.visualstudio.com/DevDiv/_git/TypeScript-VS?path=/VS/LanguageService/TypeScriptLanguageService/Features/Completion/TypeScriptCompletionService.cs&version=GBmain&line=103&lineEnd=104&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents